Userland/Applications/Escalator/EscalatorWindow.cpp

```diff
@@ -54,12 +54,17 @@ EscalatorWindow::EscalatorWindow(StringView executable, Vector<StringView> argum
     m_icon_image_widget->set_bitmap(app_icon.bitmap_for_size(32));
 
     m_ok_button = *main_widget->find_descendant_of_type_named<GUI::DialogButton>("ok_button");
-    m_ok_button->on_click = [this](auto) {
-        auto result = check_password();
-        if (result.is_error()) {
-            GUI::MessageBox::show_error(this, ByteString::formatted("Failed to execute command: {}", result.error()));
+    size_t password_attempts = 0;
+    m_ok_button->on_click = [this, password_attempts](auto) mutable {
+        if (check_password()) {
+            MUST(execute_command());
```

Comment:
That means that `Escalator` will crash if we fail to execute the command.
We should still manage the error here (as we used to do).

Reply:
I wasn't sure what error case needs to be caught, or was ever hit in the old code. I tried it with error checking and printing and error but it never printed in either of these cases.
- an executable that doesn't exist (there is a check in main whether a file exists)
- a file that is not executable (this just crashed and *prints* `posix_spawn exec: Exec format error`

But as that is likely not exhaustive, I'll chuck the error checking back in :)

Reply:
It starts to be convoluted, but if the check is in main, by exploiting a TOCTOU issue, you can make the app crash by removing the executable after it passes the check.
Anyway, let's not modify the behavior as a side effect of this patch. Also, can you fix up the commit you added into the first one?

Reply:
Ah yeh, sorry I should have thought of the timing issue as I do work in security lol. My head's not really working as usual these days though. Will fix up